apiary-metastore-listener/src/main/java/com/expedia/apiary/extensions/metastore/listener/ApiarySnsListener.java

```diff
@@ -53,21 +58,28 @@
 
   private static final Logger log = LoggerFactory.getLogger(ApiarySnsListener.class);
 
+  static final String PROTOCOL_VERSION = "1.0";
+
   private static final String TOPIC_ARN = System.getenv("SNS_ARN");
-  final static String PROTOCOL_VERSION = "1.0";
-  private final String protocolVersion = PROTOCOL_VERSION;
+
+  private final String tableParamFilter = System.getenv("TABLE_PARAM_FILTER");
+  private final Pattern tableParamFilterPattern;
 
   private final AmazonSNS snsClient;
 
   public ApiarySnsListener(Configuration config) {
-    super(config);
-    snsClient = AmazonSNSClientBuilder.defaultClient();
-    log.debug("ApiarySnsListener created ");
+    this(config, AmazonSNSClientBuilder.defaultClient());
   }
 
   ApiarySnsListener(Configuration config, AmazonSNS snsClient) {
     super(config);
     this.snsClient = snsClient;
+    tableParamFilterPattern = StringUtils.isEmpty(tableParamFilter) ? null : Pattern.compile(tableParamFilter);
```

Comment:
so I think what @massdosage  means is , either do `tableParamFilter == null ? null: Pattern.compile(tableParamFilter);` or make a test for empty string. Because this logic no transforms an empty string to null. I am not sure if that's really what you intent

Reply:
I have commented above:

> I have added a unit test for this scenario as well. I could have also used StringUtils.isBlank() to just ignore blank regex but I preferred to treat it just like any other regex as its a valid regex. I have added a log statement in the constructor to print the TABLE_PARAM_FILTER regex for faster discovery of any such issue.

Are you saying, we should treat "   " and "" in the same manner and hence use `.isBlank()` instead of `.isEmpty()` ?

Reply:
you're not treating like any other regex, you're doing 
```if regex ==""
pattern = null
```
Which means you're treating it like null.
I'm saying just check for null and any other value goes into the Pattern.compile

Reply:
Yes, I thought there was agreement today that we wouldn't filter out empty strings but the above code is still filtering them out.

Reply:
Alright. This is changed.